src/main/java/org/generationcp/middleware/service/api/study/StudyEntryService.java

```diff
@@ -29,9 +30,9 @@
 
 	StudyEntryDto replaceStudyEntry(int studyId, int entryId, int gid, String crossExpansion);
 
-	void updateStudyEntryProperty(int studyId, StudyEntryPropertyData studyEntryPropertyData);
+	void updateStudyEntriesProperty(StudyEntryPropertyDataUpdateRequestDto studyEntryPropertyDataUpdateRequestDto);
 
-	Boolean hasUnassignedEntries(int studyId);
+	List<Integer> hasPlotEntries(List<Integer> entryIds);
```

Comment:
The ACs say to check that study has experimental units. So we should implement a method under StudyService: `hasObservationUnits(final Integer studyId)`
...
```
public boolean hasObservationUnits(final Integer studyId){
  final Integer plotDatasetId = this.getPlotDatasetId(studyId);
  return this.daoFactory.getExperimentDao().count(plotDatasetId) > 0;
}
```